Fails to compile 2023-02-21

Bacon fails to compile.

# Expected behavior

It compiles and installs.

# Actual behavior

```
home@daniel-desktop3:~$ rustc --version
rustc 1.69.0-nightly (246eae2fa 2023-02-21)
home@daniel-desktop3:~$ cargo --version
cargo 1.69.0-nightly (17b3d0de0 2023-02-17)
home@daniel-desktop3:~$ cargo install-update -f bacon
    Updating registry 'https://github.com/rust-lang/crates.io-index'

Package  Installed  Latest  Needs update
bacon    v2.5.0     v2.6.0  Yes

Updating bacon
    Updating crates.io index
  Installing bacon v2.6.0
  Downloaded camino v1.1.3
  Downloaded 1 crate (32.0 KB) in 0.18s
   Compiling libc v0.2.139
   Compiling autocfg v1.1.0
   Compiling proc-macro2 v1.0.51
   Compiling quote v1.0.23
   Compiling unicode-ident v1.0.6
   Compiling cfg-if v1.0.0
   Compiling syn v1.0.107
   Compiling memchr v2.5.0
   Compiling once_cell v1.17.1
   Compiling scopeguard v1.1.0
   Compiling thiserror v1.0.38
   Compiling smallvec v1.10.0
   Compiling parking_lot_core v0.9.7
   Compiling crossbeam-utils v0.8.14
   Compiling regex-automata v0.1.10
   Compiling hashbrown v0.12.3
   Compiling crossbeam-queue v0.3.8
   Compiling quick-error v2.0.1
   Compiling rayon-core v1.10.2
   Compiling itoa v1.0.5
   Compiling parking_lot_core v0.8.6
   Compiling crc32fast v1.3.2
   Compiling adler v1.0.2
   Compiling bitflags v1.3.2
   Compiling instant v0.1.12
   Compiling same-file v1.0.6
   Compiling hex v0.4.3
   Compiling either v1.8.1
   Compiling time-core v0.1.0
   Compiling miniz_oxide v0.6.2
   Compiling human_format v1.0.3
   Compiling bytesize v1.1.0
   Compiling walkdir v2.3.2
   Compiling time-macros v0.2.7
   Compiling signal-hook v0.3.15
   Compiling sha1_smol v1.0.0
   Compiling num_threads v0.1.6
   Compiling version_check v0.9.4
   Compiling minimal-lexical v0.2.1
   Compiling log v0.4.17
   Compiling fastrand v1.9.0
   Compiling remove_dir_all v0.5.3
   Compiling serde_derive v1.0.152
   Compiling serde v1.0.152
   Compiling lock_api v0.4.9
   Compiling memoffset v0.7.1
   Compiling crossbeam-epoch v0.9.13
   Compiling num-traits v0.2.15
   Compiling hashbrown v0.13.2
   Compiling rustversion v1.0.11
   Compiling tinyvec_macros v0.1.1
   Compiling ryu v1.0.12
   Compiling unicode-bidi v0.3.10
   Compiling tinyvec v1.6.0
   Compiling percent-encoding v2.2.0
   Compiling flate2 v1.0.25
   Compiling ahash v0.8.3
   Compiling proc-macro-error-attr v1.0.4
   Compiling crossbeam-channel v0.5.6
   Compiling bstr v1.3.0
   Compiling nom v7.1.3
   Compiling form_urlencoded v1.1.0
   Compiling proc-macro-error v1.0.4
   Compiling num-integer v0.1.45
   Compiling unicode-bom v1.1.4
   Compiling static_assertions v1.1.0
   Compiling arrayvec v0.7.2
   Compiling indexmap v1.9.2
   Compiling git-bitmap v0.2.2
   Compiling aho-corasick v0.7.20
   Compiling semver v1.0.16
   Compiling crossbeam-deque v0.8.2
   Compiling camino v1.1.3
   Compiling home v0.5.4
   Compiling serde_json v1.0.93
   Compiling git-glob v0.5.4
   Compiling git-command v0.2.4
   Compiling uluru v3.0.0
   Compiling clru v0.6.1
   Compiling crossbeam v0.8.2
   Compiling regex-syntax v0.6.28
   Compiling tokio v1.25.0
   Compiling anyhow v1.0.69
   Compiling arc-swap v1.6.0
error[E0282]: type annotations needed
   --> /home/home/.cargo/registry/src/github.com-1ecc6299db9ec823/git-glob-0.5.4/src/wildmatch.rs:248:62
    |
248 | ...                   let class = &pattern.as_ref()[p_idx + 2..closing_bracket_idx - 1];
    |                                            ^^^^^^
    |
help: try using a fully qualified path to specify the expected types
    |
248 |                                         let class = &<BStr as AsRef<T>>::as_ref(pattern)[p_idx + 2..closing_bracket_idx - 1];
    |                                                      +++++++++++++++++++++++++++       ~

   Compiling os_str_bytes v6.4.1
   Compiling heck v0.4.1
For more information about this error, try `rustc --explain E0282`.
error: could not compile `git-glob` due to previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `bacon v2.6.0`, intermediate artifacts can be found at `/tmp/cargo-installIUwmjw`


Updated 0 packages.
Failed to update bacon.

home@daniel-desktop3:~$ 
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,7 +20,7 @@ coolor = "=0.5.0"
 crokey = "0.4.1"
 crossbeam = "0.8.2"
 directories-next = "2.0.0"
-git-repository = "0.30.2"
+gix = "0.37.1"
 lazy-regex = "2.3.1"
 notify = "5.0.0"
 serde = { version = "1.0.136", features = ["derive"] }

--- a/src/ignorer.rs
+++ b/src/ignorer.rs
@@ -3,7 +3,7 @@ use {
         Context,
         Result,
     },
-    git_repository::{
+    gix::{
         self as git,
         prelude::FindExt,
         Repository,

